File: Source/GortGUIv2/Gort/gortproject/templates/SampleGortProject/heuristics/DynamicLocationHeuristic.java
import gort

public class DynamicLocationHeuristic {
    public static String[] PERMISSIONS = {
        ACCESS_FINE_LOCATION,
        INTERNET
    };
    
    public static String HIT_HEADER = "screenshot2";
    
    private boolean result;
    
    @Override
    public void init(APK apk) {
        result = apk.hasPermissions(PERMISSIONS);
    }
    
    @Override
    public void onTaintLog(Taint t) {
        if (taint.matchesTaint(Taint.TAINT_LOCATION)) {
            appendHITInput(t.lastScreenshot());
        }
    }
    
    @Override
    public boolean finalInput() {
        return result;
    }
}

File: Source/GortGUIv2/Gort/gortproject/templates/SampleGortProject/heuristics/DynamicLocationHeuristic.py
class DynamicLocationHeuristic(DynamicHeuristic):
    
    PERMISSIONS = [ACCESS_FINE_LOCATION, INTERNET]
    
    HIT_HEADER = "screenshot2"
    
    def init(apk):
        self.__result = false
        result = apk.has_permissions(self.PERMISSIONS)
        
    def on_taint_log(self, taint):
        if taint.matches_taint(Taint.TAINT_LOCATION):
            append_hit_input(taint.last_screenshot())

    def final_output():
        return result